### ISSUE TITLE
[ERROR]: Cluster has memberships.hub.gke.io CRD but no  identity provider specified. 

### ISSUE THREAD

**Issue by richiefrich** [terraform-google-modules/terraform-google-kubernetes-engine#1162]
### TL;DR

I am trying to upgrade this module from 17.3.0 to 19.0.0 but I get the following error on my Terraform run. I am not sure what I am missing. 

`[](https://jenkins.ssppbuildd.zebra.engineering/blue/organizations/jenkins/phoenix-infrastructure/detail/feature%2FZRPSW-7943/10/pipeline/#step-23-log-2317)[](https://jenkins.ssppbuildd.zebra.engineering/blue/organizations/jenkins/phoenix-infrastructure/detail/feature%2FZRPSW-7943/10/pipeline/#step-23-log-2318)[](https://jenkins.ssppbuildd.zebra.engineering/blue/organizations/jenkins/phoenix-infrastructure/detail/feature%2FZRPSW-7943/10/pipeline/#step-23-log-2319)[](https://jenkins.ssppbuildd.zebra.engineering/blue/organizations/jenkins/phoenix-infrastructure/detail/feature%2FZRPSW-7943/10/pipeline/#step-23-log-2320)[](https://jenkins.ssppbuildd.zebra.engineering/blue/organizations/jenkins/phoenix-infrastructure/detail/feature%2FZRPSW-7943/10/pipeline/#step-23-log-2321)[](https://jenkins.ssppbuildd.zebra.engineering/blue/organizations/jenkins/phoenix-infrastructure/detail/feature%2FZRPSW-7943/10/pipeline/#step-23-log-2322)asm_kubeconfig get memberships.hub.gke.io -ojsonpath={..metadata.name}'

install_asm_1.10: -------------

install_asm_1.10: [ERROR]: Cluster has memberships.hub.gke.io CRD but no

identity provider specified. Please ensure that

an identity provider is available for the registered cluster.

+ cleanup

+ rm -rf /tmp/kubectl_wrapper_23091_17865`

Here are the versions I am currently using. 
`    google = {
      source  = "hashicorp/google"
      version = "4.12.0"
    }
    google-beta = {
      source  = "hashicorp/google-beta"
      version = "4.12.0"
    }`

Then I have this as code for ASM module. 
`module "asm-primary" {
  source                    = "terraform-google-modules/kubernetes-engine/google//modules/asm"
  version                   = "19.0.0"
  asm_version               = var.asm_version
  project_id                = var.project_id
  enable_all                = true
  managed_control_plane     = true
  cluster_endpoint          = module.primary-cluster.endpoint
  cluster_name              = module.primary-cluster.name
  location                  = module.primary-cluster.location
  enable_cluster_labels     = true
  enable_gcp_apis           = true
  enable_gcp_iam_roles      = false
  enable_gcp_components     = true
  skip_validation           = false
}`
